src/main/java/jewellery/inventory/service/ResourceInOrganizationService.java

```diff
@@ -0,0 +1,199 @@
+    logger.debug(
+        "Finding resource by ID: {}, Finding organization by ID: {}",
+        resourceId,
+        organization.getId());
+    return organization.getResourceInOrganization().stream()
+        .filter(resource -> resource.getResource().getId().equals(resourceId))
+        .findFirst();
+  }
+
+  private ResourceInOrganization createAndAddNewResourceInOrganization(
+      Organization organization, Resource resource, BigDecimal quantity) {
+    ResourceInOrganization resourceInOrganization =
+        ResourceInOrganization.builder()
+            .organization(organization)
+            .resource(resource)
+            .quantity(quantity)
+            .build();
+    logger.info("New resource in organization created: {}", resourceInOrganization);
+    return resourceInOrganization;
+  }
+
+  private ResourceInOrganization getResourceInOrganization(
+      Organization organization, Resource resource) {
+    logger.debug(
+        "Getting resource in organization. Organization: {}, Resource: {}", organization, resource);
+    return findResourceInOrganization(organization, resource.getId())
+        .orElseGet(
+            () -> createAndAddNewResourceInOrganization(organization, resource, BigDecimal.ZERO));
+  }
+
+  private ResourceInOrganization addResourceToOrganization(
+      Organization organization, Resource resource, BigDecimal quantity, BigDecimal dealPrice) {
+    logger.info(
+        "Adding resource to organization. Organization: {}, Resource: {}, Quantity: {}",
+        organization,
+        resource,
+        quantity);
+    ResourceInOrganization resourceInOrganization =
+        getResourceInOrganization(organization, resource);
+    resourceInOrganization.setQuantity(resourceInOrganization.getQuantity().add(quantity));
+    resourceInOrganization.setDealPrice(dealPrice);
+    resourceInOrganizationRepository.save(resourceInOrganization);
+    logger.debug("ResourceInOrganization after addition: {}", resource);
+    return resourceInOrganization;
+  }
+
+  @Override
+  public Object fetchEntity(Object... ids) {
+    if (ids != null && ids.length > 0) {
+      if (ids[0] instanceof ResourceInOrganizationRequestDto resourceInOrganizationRequestDto) {
+
+        return getResourceInOrganizationResponse(
+            resourceInOrganizationRequestDto.getOrganizationId(),
+            resourceInOrganizationRequestDto.getResourceId());
+      } else {
+        return getResourceInOrganizationResponse((UUID) ids[0], (UUID) ids[1]);
+      }
+    }
+    return null;
+  }
+
+  private ResourcesInOrganizationResponseDto getResourceInOrganizationResponse(
+      UUID organizationId, UUID resourceId) {
+    ResourceInOrganization resourceInOrganization =
+        getResourceInOrganization(organizationId, resourceId);
+    if (resourceInOrganization != null) {
+      return resourceInOrganizationMapper.toResourcesInOrganizationResponse(resourceInOrganization);
+    }
+    return null;
```

Comment:
This check shouldn't be needed if we've moved the null check in the mapper.

Reply:
Done!